File: src/tauri/commands/ai-task.ts
import { safeInvoke } from "./_base";

// ── ai_tasks ────────────────────────────────────────────────────────────

export type AiTaskCategory = "screen" | "batch" | "part" | "other";

export type AiTaskResult = {
  id: number;
  task_cd: string;
  task_name: string;
  category: string;
  is_complete: boolean;
  completed_at: string;
  created_at: string;
  created_by: string;
  updated_at: string;
  updated_by: string;
};

export type CreateTaskRequest = {
  task_cd: string;
  task_name: string;
  category: string;
};

export type UpdateTaskRequest = {
  task_cd: string;
  task_name: string;
  category: string;
  is_complete: boolean;
};

export function aiTaskCreate(username: string, request: CreateTaskRequest) {
  return safeInvoke<AiTaskResult>("ai_task_create", { username, request });
}

export function aiTaskList(keyword?: string, isComplete?: boolean) {
  return safeInvoke<AiTaskResult[]>("ai_task_list", {
    keyword: keyword ?? null,
    isComplete: isComplete ?? null,
  });
}

export function aiTaskUpdate(id: number, username: string, request: UpdateTaskRequest) {
  return safeInvoke<AiTaskResult>("ai_task_update", { id, username, request });
}

// ── ai_task_wf_proc ─────────────────────────────────────────────────────

export type AiTaskWfProcResult = {
  id: number;
  task_id: number;
  wf_id: number;
  latest_step_id: number | null;
  created_at: string;
  created_by: string;
  updated_at: string;
  updated_by: string;
};

export type CreateWfProcRequest = {
  task_id: number;
  wf_id: number;
};

export function aiTaskWfProcCreate(username: string, request: CreateWfProcRequest) {
  return safeInvoke<AiTaskWfProcResult>("ai_task_wf_proc_create", { username, request });
}

export function aiTaskWfProcList(taskId: number) {
  return safeInvoke<AiTaskWfProcResult[]>("ai_task_wf_proc_list", { taskId });
}

export function aiTaskWfProcUpdate(id: number, latestStepId: number, username: string) {
  return safeInvoke<AiTaskWfProcResult>("ai_task_wf_proc_update", { id, latestStepId, username });
}

// ── ai_task_wf_proc_step ────────────────────────────────────────────────

export type WfProcStepStatus = "pending" | "in_progress" | "completed" | "skipped";

export type AiTaskWfProcStepResult = {
  id: number;
  wf_proc_id: number;
  wf_step_id: number;
  status: string;
  created_at: string;
  created_by: string;
  updated_at: string;
  updated_by: string;
};

export type CreateWfProcStepRequest = {
  wf_proc_id: number;
  wf_step_id: number;
  status: string;
};

export function aiTaskWfProcStepCreate(username: string, request: CreateWfProcStepRequest) {
  return safeInvoke<AiTaskWfProcStepResult>("ai_task_wf_proc_step_create", { username, request });
}

export function aiTaskWfProcStepList(wfProcId: number) {
  return safeInvoke<AiTaskWfProcStepResult[]>("ai_task_wf_proc_step_list", { wfProcId });
}

export function aiTaskWfProcStepUpdate(id: number, status: string, username: string) {
  return safeInvoke<AiTaskWfProcStepResult>("ai_task_wf_proc_step_update", { id, status, username });
}

File: src-tauri/src/models/ai_task.rs
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct AiTask {
    pub id: i32,
    pub task_cd: String,
    pub task_name: String,
    pub category: String,
    pub is_complete: bool,
    pub completed_at: String,
    pub created_at: String,
    pub created_by: String,
    pub updated_at: String,
    pub updated_by: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateTaskRequest {
    pub task_cd: String,
    pub task_name: String,
    pub category: String,
}

#[derive(Debug, Deserialize)]
pub struct UpdateTaskRequest {
    pub task_cd: String,
    pub task_name: String,
    pub category: String,
    pub is_complete: bool,
}

#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct AiTaskWfProc {
    pub id: i32,
    pub task_id: i32,
    pub wf_id: i32,
    pub latest_step_id: Option<i32>,
    pub created_at: String,
    pub created_by: String,
    pub updated_at: String,
    pub updated_by: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateWfProcRequest {
    pub task_id: i32,
    pub wf_id: i32,
}

#[derive(Clone, Debug, Serialize, Deserialize)]
pub struct AiTaskWfProcStep {
    pub id: i32,
    pub wf_proc_id: i32,
    pub wf_step_id: i32,
    pub status: String,
    pub created_at: String,
    pub created_by: String,
    pub updated_at: String,
    pub updated_by: String,
}

#[derive(Debug, Deserialize)]
pub struct CreateWfProcStepRequest {
    pub wf_proc_id: i32,
    pub wf_step_id: i32,
    pub status: String,
}
